tutu-backend/src/test/scala/fi/oph/tutu/backend/ContollerUnit.scala

```diff
@@ -0,0 +1,133 @@
+package fi.oph.tutu.backend
+
+import fi.oph.tutu.backend.repository.HakemusRepository
+import fi.oph.tutu.backend.controller.Controller
+import fi.oph.tutu.backend.controller.Hakemus
+import fi.oph.tutu.backend.service.*
+import fi.oph.tutu.backend.domain.*
+import fi.oph.tutu.backend.utils.{AuditLog, AuthoritiesUtil}
+import fi.oph.tutu.backend.security.SecurityConstants
+import org.junit.jupiter.api.*
+import org.junit.jupiter.api.MethodOrderer.OrderAnnotation
+import org.junit.jupiter.api.TestInstance.Lifecycle
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.*
+import org.springframework.beans.factory.annotation.{Autowired, Qualifier}
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest
+import org.springframework.http.HttpStatus
+import org.springframework.security.test.context.support.WithMockUser
+import org.springframework.security.test.web.servlet.setup.SecurityMockMvcConfigurers
+import org.springframework.test.context.ActiveProfiles
+import org.springframework.test.context.bean.`override`.mockito.MockitoBean
+import org.springframework.test.web.servlet.MockMvc
+import org.springframework.test.web.servlet.assertj.*
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers.*
+import org.springframework.test.web.servlet.setup.{DefaultMockMvcBuilder, MockMvcBuilders, MockMvcConfigurer}
+
+import java.util.UUID
+
+
+// TODO: Mockito Qualifiers!
+
+@WebMvcTest(controllers = Array(classOf[Controller]))
+class ControllerUnitTest {
+
+  @MockitoBean
+  private val hakemuspalveluService : HakemuspalveluService = null
+
+  @MockitoBean
+  private val userService : UserService = null
```

Comment:
Nämä servicet mitä ei testeissä tarvitse mutta joita tarvitaan tietylle servicelle, voidaan mockata näin: 

https://github.com/Opetushallitus/ovara-virkailija/blob/e041ff384ee0fecfede9d0f289c4f181b14225f0/ovara-backend/src/test/scala/fi/oph/ovara/backend/CommonServiceTest.scala#L12

niin ei tartte ylimääräisiä when(....) -lauseita.